modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -1956,13 +1959,10 @@ def unique(self, keep="first", ignore_index=True, subset=None):
                 ),
                 preserve_columns=True,
             )
+            return self.__constructor__(new_modin_frame, shape_hint=self._shape_hint)
         else:
-            new_modin_frame = self._modin_frame.apply_full_axis(
-                0,
-                lambda x: x.squeeze(axis=1).unique(),
-                new_columns=self.columns,
-            )
-        return self.__constructor__(new_modin_frame, shape_hint=self._shape_hint)
+            res = self.to_pandas().squeeze(axis=1)
+            return res.unique()
```

Comment:
@dchigarev I measured the new approach on small data (using your script from https://github.com/modin-project/modin/pull/7091). It seems to work faster than the previous one. Do you have the ability to measure on big sizes? (maybe you still have your env)
<img width="563" alt="image" src="https://github.com/modin-project/modin/assets/45976948/8ff205a7-19cc-45f3-b9fb-ff35780065d7">

Reply:
I think it will always work faster, since the old full-column approach is basically '.to_pandas() + splitting + putting splits to ray storage', and in your approach it's only .to_pandas().

Reply:
> I think it will always work faster, since the old full-column approach is basically '.to_pandas() + splitting + putting splits to ray storage', and in your approach it's only .to_pandas().

Yes, but I compare with range-partitioning implementation.

Reply:
Running the same scripts on linux gives different results (range-partitioning is still faster in a lot of cases):

<details><summary>perf results for linux</summary>

MODIN_CPUS=8
![image](https://github.com/modin-project/modin/assets/62142979/389ae9ed-f794-45f7-be03-92a5e35826b6)

MODIN_CPUS=16
![image](https://github.com/modin-project/modin/assets/62142979/c8da7ff2-e1ba-4fbb-9df8-78563b88ab54)

</details>

I've been able to reproduce your results though when tried to run the same on windows:

<details><summary>perf results for windows</summary>

![image](https://github.com/modin-project/modin/assets/62142979/e4f2d685-f727-48ac-af9c-40be8fc2c403)

</details>

I don't think that the main reasons for the difference is OS here. I think that it's a hardware difference, since for windows runs I've used my own laptop instead of a dedicated machine.

Reply:
@dchigarev thank you very much for running the benchmark!

> I don't think that the main reasons for the difference is OS here. I think that it's a hardware difference, since for windows runs I've used my own laptop instead of a dedicated machine.

Interesting detail.